### DISCUSSION TITLE
Returning the data from a service!

### DISCUSSION THREAD

**Discussion by bvbasavaraju** [intel/compile-time-init-build#183] [Q&A] [answered]
We can pass in 0 or more number of parameter to service implementation! is there any way to return the value/ref of some data from service implementation!!??

**Comment by lukevalenty** [comment, marked as answer]
It can be done, but not with the cib::callback service. You would need to implement your own service that knows how to gather data from all the extensions and formulate a return value.

**Reply by bvbasavaraju** [reply]
Thank you for the answer.
